### PR TITLE
Closes #2146 Fix Time Imputation Flag

### LINKED ISSUE #2146: Bug: Time imputation flag incorrect
### What happened?

In the following case the time imputation flag is not derived correctly.
```
> compute_tmf("2022-05--T00:00", ymd_hms("2022-05-15T23:59:59"))
[1] "S"
```
It should be `"H"` as the hours were changed.

### Session Information

_No response_

### Reproducible Example

_No response_

### PR DESCRIPTION
**Pull request by ahasoplakus**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_date_vars.R [hunk 1]
```diff
@@ -1207,25 +1207,41 @@ compute_dtf <- function(dtc, dt) {
...
   assert_character_vector(dtc)
   assert_logical_scalar(ignore_seconds_flag)
 
+  valid_dtc <- is_valid_dtc(dtc)
+  warn_if_invalid_dtc(dtc, valid_dtc)
+
   partial <- get_partialdatetime(dtc)
   highest_miss <- convert_blanks_to_na(vector("character", length(dtc)))
-  for (c in c("hour", "minute", "second")) {
+  hms <- c("hour", "minute", "second")
```
Comment (zdz2101) [inline R/derive_date_vars.R:1231]:
```suggestion
# concatenate lubridate functions: `hour()`, `minute()`, `second()` to map over dtm input
  hms <- c("hour", "minute", "second")
```

File: R/derive_date_vars.R [hunk 2]
```diff
@@ -1207,25 +1207,41 @@ compute_dtf <- function(dtc, dt) {
...
+  time_part <-
```
Comment (zdz2101) [inline R/derive_date_vars.R:1232]:
```suggestion
# extract hour, minute, second over each value of dtm and put into a list time_part
  time_part <-
```

File: R/derive_date_vars.R [hunk 3]
```diff
@@ -1207,25 +1207,41 @@ compute_dtf <- function(dtc, dt) {
...
+    map(set_names(hms), \(y) map_dbl(dtm, \(x) exec(y, x)))
```
Comment (zdz2101) [inline R/derive_date_vars.R:1233]: I really like this! It is really clever but certainly has a different look and flavor to rest of our codebase, so I left some recommended comments for the code. In addition, support of the new anonymous function short hand was introduced in purrr 1.0.0, so can you add a change to the DESCRIPTION file to include `purrr (>= 1.0.0)` ?

**Review by zdz2101 — CHANGES_REQUESTED** [review]
Thanks for your contribution, mostly just small stuff remiaing: modify `DESCRIPTION` to account for `purrr 1.0.0` features, and make an entry for `NEWS`/changelog and this looks almost ready to go!

File: R/derive_date_vars.R (hunk 3 above)
Comment (ddsjoberg) [reply R/derive_date_vars.R:1233]: I think this is an R 4.1 addition, rather than purrr 🐱

Comment (zdz2101) [reply R/derive_date_vars.R:1233]: Looks like a modification to `Depends: R (>= 4.1)` is necessary too!

Comment (ddsjoberg) [reply R/derive_date_vars.R:1233]: In this case, I would just update the line to 

```r
map(set_names(hms), function(y) map_dbl(dtm, function(x) exec(y, x)))
```

so we don't make an R version dependency for something that is not needed.

File: R/derive_date_vars.R (hunk 1 above)
Comment (ahasoplakus) [reply R/derive_date_vars.R:1233]: Updated.

File: R/derive_date_vars.R (hunk 2 above)
Comment (ahasoplakus) [reply R/derive_date_vars.R:1236]: Updated.

File: R/derive_date_vars.R (hunk 3 above)
Comment (ahasoplakus) [reply R/derive_date_vars.R]: Thanks @zdz2101 @ddsjoberg for the review. I removed the anonymous function shorthand and added the suggested comments.

**Comment by bms63** [comment]
Please just update the `News.md` and this looks ready to go to me.

**Comment by ahasoplakus** [comment]
> @ahasoplakus Please just update the `News.md` and this looks ready to go to me.

Updated.

**Review by bundfussr — APPROVED** [review]

**Review by bms63 — APPROVED** [review]

**Pull request merged**